core/evm.go

```diff
@@ -29,6 +29,11 @@ import (
 	staking "github.com/harmony-one/harmony/staking/types"
 )
 
+type headerGetter interface {
```

Comment:
why refactor the ChainContext into headerGetter?   if it's not necessary, better keep as is.

Reply:
Because `NewEVMContext()` is used in tx pool's `validateTx()` and since pool's `blockChain` attribute does not support the ChainContext interface. this way is the least changes to wire it together.